File: topico_4/Python/11-intersec.py
# Recebe duas listas e retorna a intercessão entre elas

def intersec(xs, ys):
    lista = []
    for x in xs:
        for y in ys:
            if x == y:
                lista.append(x)
    return lista

# def intersec(xs, ys):
#    lista = []
#    lista = xs.intersection(ys)

print(intersec ([3], [3]) == [3]) #True
print(intersec ([3,4,1], [1,4,3]) == [3,4,1]) #True
print(intersec ([3,6,5,7], [9,7,5,1,3,6]) == [3,6,5,7]) #True
print(intersec ([3,6,5,7], [9,7,5,1,3]) == [3,5,7]) #True

File: topic_4/Haskell/11-intersec.hs
-- Recebe duas listas e retorna a intercessão entre elas
elemento x [] = False -- retorna se um elemento está na lista
elemento x (y:ys) 
    | x == y = True
    | otherwise = elemento x ys

intersec [] [] = []
intersec xs [] = []
intersec [] xs = []
intersec xs ys = [x | x <- xs, elemento x ys]

-- Também posso fazer assim:
-- intersec (x:xs) ys = if(elemento x ys) then x : intersec xs ys else intersec xs ys

main = do
    a <- readLn :: IO [Int]
    b <- readLn :: IO [Int]
    print $ intersec a b